scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+            local petArg = playerArg:getPet()
+
+            if petArg and petArg:getMobMod(xi.mobMod.MAGIC_COOL) > 1 then
+                setMagicCastCooldown(petArg)
+            end
+        end)
+
+        master:addListener('ABILITY_USE', playerListenerVar .. 'ABILITY', function(playerArg, target, ability, action)
+            local petArg = playerArg:getPet()
+            local abilityName = string.lower(ability:getName())
+
+            if
+                petArg and
+                (abilityName == 'assault' or
+                abilityName == 'retreat')
+            then
+                -- reset cast cooldown buffer and timer
+                local lastCastTime = pet:getLocalVar(lastCastTimeVar)
+                local lastCastTimeStamp = pet:getLocalVar(lastCastTimeStampVar)
+                if
+                    lastCastTimeStamp > 0
+                then
+                    lastCastTime = lastCastTime + os.time() - lastCastTimeStamp
+                end
+
+                pet:setLocalVar(lastCastTimeStampVar, 0)
+                pet:setLocalVar(lastCastTimeVar, lastCastTime)
+                pet:setLocalVar(buffModeVar, 1)
+                setMagicCastCooldown(petArg)
+            end
+        end)
+    end
+end
+
+xi.pets.avatar.onMobDeath = function(pet)
+    local master = pet:getMaster()
+
+    if master and master:getObjType() == xi.objType.PC then
+        master:removeListener(playerListenerVar)
+        master:removeListener(playerListenerVar .. 'ABILITY')
+    end
+end
+
+xi.pets.avatar.onMobMagicPrepare = function(pet)
+    -- Note that:
+    -- returning -1 in this function forces TryCastSpell to exit without casting a spell, but
+    -- will still set the m_LastMagicTime to ensure next call of this function is after the cast delay
+    -- Also, if we return nothing TryCastSpell will default to normal mob casting behavior (nukes from spell list, etc)
+-- print('onMobMagicPrepare: ' .. pet:getMobMod(xi.mobMod.MAGIC_COOL))
+    local master = pet:getMaster()
+
+    if master and master:getObjType() == xi.objType.PC then
+        -- meta checks for fresh pet, etc
+        pet:setLocalVar(lastCastTimeVar, 0)
+        pet:setLocalVar(lastCastTimeStampVar, os.time())
+        local spellID = 0
+        local spellTarget = nil
+        if pet:getMobMod(xi.mobMod.MAGIC_COOL) == 1 then
+            setMagicCastCooldown(pet)
+
+            return -1
+        end
+
+        pet:setLocalVar(buffModeVar, 1)
+
+        -- Core functionality to decide which spell to use
+        spellID, spellTarget = xi.pets.avatar.getSpiritSpell(pet)
+
+        -- Final items to cast the spell and ensure cast delay is proper
+        local spell = GetSpell(spellID)
+        if spell then
+            if spell:getSkillType() == xi.skill.ENHANCING_MAGIC then
+                -- half casting delay
+                pet:setLocalVar(buffModeVar, 0)
+            end
+-- print(spellTarget:getName() .. '->' .. spellID)
```

Comment:
Can remove commented out prints, or if this is likely to need to be debugged/tweaked/tuned at some point: replace with a local debug printer, similar to here: 
https://github.com/LandSandBoat/server/blob/base/scripts/globals/chocobo_racing.lua#L19-L25

Reply:
I like it, done